source/MaterialXCore/Element.h

```diff
@@ -761,14 +761,16 @@ class MX_CORE_API Element : public std::enable_shared_from_this<Element>
     /// name, and attributes.
     string asString() const;
 
-    /// Resolve a reference to a named element at the root scope of this document,
+    /// Resolve a reference to a named element under a given root. If
+    /// If no root is provided then the root scope of this document is used,
     /// taking the namespace at the scope of this element into account.
-    template <class T> shared_ptr<T> resolveRootNameReference(const string& name) const
+    template <class T> shared_ptr<T> resolveRootNameReference(const string& name, ConstElementPtr root = nullptr) const
```

Comment:
Now that this method supports name resolution at arbitrary scopes, we should likely choose a new method name that isn't restricted to root scope.  Perhaps something along the lines of the following:

```
/// Resolve a reference to a named element at the scope of the given parent,
/// taking the namespace at the scope of this element into account.  If no parent
/// is provided, then the root scope of the document is used.
template <class T> shared_ptr<T> resolveNameReference(const string& name, ConstElementPtr parent= nullptr) const
```

Reply:
This sounds good. I've made the change. This makes me think to keep a note to see if other places can be made more robust by not always assuming root.

Reply:
Hi @ix-dcourtois (Damien)
Yes, this is something that has come up before and worth discussing as I also come from a history of it being "taboo" to break API compatibility :). Let's open an issue. 

@jstone-lucasfilm, I guess we can start with what you see as the precedence for API compatibility.

BTW, for this case, it will not "break" as this method was never meant to be public and is now renamed and hidden.

Reply:
@ix-dcourtois Completely agreed that changing the signatures of methods in the public API should be done as rarely as possible, since it has the potential to impact so many users.  In this case, we're considering the change because `resolveRootNameReference` never quite made sense as a public API method, and it's my sense that it was swept into the Python/JS bindings for completeness, rather than being a method that developers would want to use.  If you're aware of use cases for this method in developer code, let us know; otherwise we're thinking of making this a "protected" method that would only be visible within the MaterialX libraries themselves.

Reply:
No complaints for me, I was just pointing out the issue, we're not actually using this one in our code (we had the problem with other methods in which parameters were added or removed, I don't remember, but this one going to protected is ok, at least for us)